src/feature/hs/hs_client.c

```diff
@@ -1612,6 +1612,90 @@ get_creds_from_client_auth_filename(const char *filename,
   return auth;
 }
 
+/*
+ * Remove the file in <b>filename</b> under the global client auth credential
+ * storage.
+ */
+static void
+remove_client_auth_creds_file(const char *filename)
+{
+  char *creds_file_path = NULL;
+  const or_options_t *options = get_options();
+
+  creds_file_path = hs_path_from_filename(options->ClientOnionAuthDir,
+                                          filename);
+  if (file_status(creds_file_path) != FN_FILE) {
+    log_warn(LD_REND, "Failed to remove client auth file (Not a file) (%s).",
+             creds_file_path);
+    goto end;
+  }
+
+  if (tor_unlink(creds_file_path) != 0) {
+    log_warn(LD_REND, "Failed to remove client auth file (%s).",
+             creds_file_path);
+    goto end;
+  }
```

Comment:
Hmmm, don't we have a TOCTOU issue race? Seems possibly harmless but you never know. I would probably just let `tor_unlink()` fail here instead of looking if the file is legit before?

Reply:
Fixed in 2852f8d732.

FWIW I was following the logic from `expire_old_onion_keys()` but this new logic is also fine.